int-set/src/sparse_bit_set.rs

```diff
@@ -41,6 +41,19 @@ impl IntSet<u32> {
     /// Sparse bit sets are a specialized, compact encoding of bit sets defined in the IFT specification:
     /// <https://w3c.github.io/IFT/Overview.html#sparse-bit-set-decoding>
     pub fn from_sparse_bit_set(data: &[u8]) -> Result<IntSet<u32>, DecodingError> {
+        Self::from_sparse_bit_set_bounded(data, u32::MAX as u64 + 1)
+    }
+
+    /// Populate this set with the values obtained from decoding the provided sparse bit set bytes.
+    /// The size (in number of members) of the output set is bounded to be equal to or below the provided max size.
+    /// If the decoded set would contain more members than the limit an error will be returned.
+    ///
+    /// Sparse bit sets are a specialized, compact encoding of bit sets defined in the IFT specification:
+    /// <https://w3c.github.io/IFT/Overview.html#sparse-bit-set-decoding>
+    pub fn from_sparse_bit_set_bounded(
```

Comment:
do we desire for this to actually be public API? my hunch is no, and it's really just useful for the fuzzer, in which case I would add `#[doc(hidden)]` and include a note that it is not considered public for the purposes of semver.

Reply:
This, or something like this will be needed in the public api. For example in the IFT case we know we are decoding sets that contain unicode values so we can restrict the maximum set size accordingly to put an upper bound on sparse set decode times.

We can introduce a bound in two different ways:
1. Maximum set size.
2. Maximum value allowed to be decoded.

For now I choose 1. as that allows the fuzzer to still probe edge cases (eg. u32::MAX), but ultimately 2. might be more useful for things like IFT. My plan for now was to leave 1. public and then potentially revisit this as I work through the IFT stuff.